### ISSUE TITLE
Error Mapping NPoco Expando object to DTO

### ISSUE THREAD

**Issue by mustafajavid** [AutoMapper/AutoMapper#2516]
Hi,

I am trying to map an NPoco expando object to DTO but it gives me an exception:

```
Error mapping types.

Mapping types:
PocoExpando -> DTO
```

I tried to do it by setting `CreateMissingTypeMaps` to true and also by explicitly mentioning the types like `CreateMap<dynamic, DTO>()` but nothing is working for me.

I am using version 6.2.2

**Comment by lbargaoanu** [comment]
I think this is better suited for StackOverflow.

**Comment by mustafajavid** [comment]
Hi @lbargaoanu! Can you please help with this issue?
https://stackoverflow.com/questions/48519255/error-mapping-npoco-expando-object-to-dto-with-automapper

**Comment by mustafajavid** [comment]
@lbargaoanu I am wondering if there's no support for NPoco's `PocoExpando `object in AutoMapper as I have already tried many solutions that are discussed in different threads.

**Comment by lbargaoanu** [comment]
You'll have to remove NPoco from this discussion :) [The docs.](http://automapperdocs.readthedocs.io/en/latest/Dynamic-and-ExpandoObject-Mapping.html)

**Comment by mustafajavid** [comment]
@lbargaoanu It is not working.

I have tried following 3 approaches:

First
```
var config = new MapperConfiguration(cfg => cfg.CreateMissingTypeMaps = true);
var mapper = config.CreateMapper();
var dto = mapper.Map<DTO>(entity);
```
Second:
```
var config = new MapperConfiguration(cfg => cfg.CreateMap<dynamic, DTO>());
var mapper = config.CreateMapper();
var dto = mapper.Map<dynamic, DTO>(entity);
```

Third:
```
Mapper.Initialize(cfg => { });
var dto = Mapper.Map<DTO>(entity);
```
I am getting the same error for all above approaches

**Comment by lbargaoanu** [comment]
A repro would help. Make a [gist](https://gist.github.com/lbargaoanu/9c7233441c3a3413cc2b9b9ebb5964a9) that we can execute and see fail.

**Comment by mustafajavid** [comment]
We were previously using AutoMapper 4.0.1 version and there was no issue with this version.
This is how we had implemented in 4.0.1 version
```
var config = new ConfigurationStore(new TypeMapFactory(), MapperRegistry.Mappers);
var engine = new MappingEngine(config);
var dto = engine.Map<dynamic, DTO>(expandoObject);
```
But once, we have updated AutoMapper to 6.2.2 version this issue has started appearing.

**Comment by mustafajavid** [comment]
Sorry it was an issue with the data. Thanks for the help :)
